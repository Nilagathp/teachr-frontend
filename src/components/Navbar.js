import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

import { withStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Button from "@material-ui/core/Button";
import { clearCourse } from "../redux/actions/courseActions";
import { clearCategory } from "../redux/actions/categoryActions";
import { clearStatus } from "../redux/actions/statusActions";
import { clearValue } from "../redux/actions/valueActions";

import { logOutUser } from "../redux/actions/userActions";

const styles = theme => ({
  appBar: {
    zIndex: theme.zIndex.drawer + 1,
    root: {
      display: "flex"
    }
  }
});

class Navbar extends React.Component {
  render() {
    const { person, logOutUser, classes } = this.props;
    let name;
    person.teacher
      ? (name = person.teacher.name)
      : (name = person.student.name);
    return (
      <AppBar
        id="navbar"
        position="fixed"
        color="default"
        className={classes.appBar}
      >
        <Toolbar>
          <Link to="/home" style={{ textDecoration: "none" }}>
            <Button
              onClick={() => {
                this.props.clearValue();
                this.props.clearCategory();
                this.props.clearCourse();
                this.props.clearStatus();
              }}
              style={{
                textDecoration: "none",
                textTransform: "none",
                fontSize: 20,
                paddingLeft: 0,
                margin: 0
              }}
            >
              {name}
            </Button>
          </Link>
          {person ? (
            <Button style={{ marginLeft: "auto" }} onClick={logOutUser}>
              Logout
            </Button>
          ) : null}
        </Toolbar>
      </AppBar>
    );
  }
}

const mapStateToProps = state => {
  return { person: state.user.person };
};

export default withStyles(styles)(
  connect(
    mapStateToProps,
    { logOutUser, clearCategory, clearValue, clearCourse, clearStatus }
  )(Navbar)
);

export { Navbar };
