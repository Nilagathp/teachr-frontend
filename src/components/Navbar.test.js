import React from "react";
import { shallow } from "enzyme";

import { Navbar } from "./Navbar";

const person = { teacher: { name: "Test Teacher" } };
const classes = {
  appBar: {
    root: {
      display: "flex"
    }
  }
};

describe("Navbar", () => {
  it("renders correctly", () => {
    let wrapper = shallow(<Navbar person={person} classes={classes} />);
    expect(wrapper.find("#navbar").length).toEqual(1);
  });
  it("matches its snapshot", () => {
    let wrapper = shallow(<Navbar person={person} classes={classes} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("displays teacher's name when user is a teacher", () => {
    const component = shallow(<Navbar classes={classes} person={person} />);
    expect(component).toMatchSnapshot();
  });
});
