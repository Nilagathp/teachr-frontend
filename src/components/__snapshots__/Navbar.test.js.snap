// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar displays teacher's name when user is a teacher 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar
    classes={
      Object {
        "appBar": Object {
          "root": Object {
            "display": "flex",
          },
        },
      }
    }
    person={
      Object {
        "teacher": Object {
          "name": "Test Teacher",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Toolbar)>
        <Link
          replace={false}
          style={
            Object {
              "textDecoration": "none",
            }
          }
          to="/home"
        >
          <WithStyles(Button)
            onClick={[Function]}
            style={
              Object {
                "fontSize": 20,
                "margin": 0,
                "paddingLeft": 0,
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
          >
            Test Teacher
          </WithStyles(Button)>
        </Link>
        <WithStyles(Button)
          style={
            Object {
              "marginLeft": "auto",
            }
          }
        >
          Logout
        </WithStyles(Button)>
      </WithStyles(Toolbar)>,
      "className": Object {
        "root": Object {
          "display": "flex",
        },
      },
      "color": "default",
      "id": "navbar",
      "position": "fixed",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            style={
              Object {
                "textDecoration": "none",
              }
            }
            to="/home"
          >
            <WithStyles(Button)
              onClick={[Function]}
              style={
                Object {
                  "fontSize": 20,
                  "margin": 0,
                  "paddingLeft": 0,
                  "textDecoration": "none",
                  "textTransform": "none",
                }
              }
            >
              Test Teacher
            </WithStyles(Button)>
          </Link>,
          <WithStyles(Button)
            style={
              Object {
                "marginLeft": "auto",
              }
            }
          >
            Logout
          </WithStyles(Button)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Button)
              onClick={[Function]}
              style={
                Object {
                  "fontSize": 20,
                  "margin": 0,
                  "paddingLeft": 0,
                  "textDecoration": "none",
                  "textTransform": "none",
                }
              }
            >
              Test Teacher
            </WithStyles(Button)>,
            "replace": false,
            "style": Object {
              "textDecoration": "none",
            },
            "to": "/home",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Test Teacher",
              "onClick": [Function],
              "style": Object {
                "fontSize": 20,
                "margin": 0,
                "paddingLeft": 0,
                "textDecoration": "none",
                "textTransform": "none",
              },
            },
            "ref": null,
            "rendered": "Test Teacher",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Logout",
            "onClick": undefined,
            "style": Object {
              "marginLeft": "auto",
            },
          },
          "ref": null,
          "rendered": "Logout",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Toolbar)>
          <Link
            replace={false}
            style={
              Object {
                "textDecoration": "none",
              }
            }
            to="/home"
          >
            <WithStyles(Button)
              onClick={[Function]}
              style={
                Object {
                  "fontSize": 20,
                  "margin": 0,
                  "paddingLeft": 0,
                  "textDecoration": "none",
                  "textTransform": "none",
                }
              }
            >
              Test Teacher
            </WithStyles(Button)>
          </Link>
          <WithStyles(Button)
            style={
              Object {
                "marginLeft": "auto",
              }
            }
          >
            Logout
          </WithStyles(Button)>
        </WithStyles(Toolbar)>,
        "className": Object {
          "root": Object {
            "display": "flex",
          },
        },
        "color": "default",
        "id": "navbar",
        "position": "fixed",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
              to="/home"
            >
              <WithStyles(Button)
                onClick={[Function]}
                style={
                  Object {
                    "fontSize": 20,
                    "margin": 0,
                    "paddingLeft": 0,
                    "textDecoration": "none",
                    "textTransform": "none",
                  }
                }
              >
                Test Teacher
              </WithStyles(Button)>
            </Link>,
            <WithStyles(Button)
              style={
                Object {
                  "marginLeft": "auto",
                }
              }
            >
              Logout
            </WithStyles(Button)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Button)
                onClick={[Function]}
                style={
                  Object {
                    "fontSize": 20,
                    "margin": 0,
                    "paddingLeft": 0,
                    "textDecoration": "none",
                    "textTransform": "none",
                  }
                }
              >
                Test Teacher
              </WithStyles(Button)>,
              "replace": false,
              "style": Object {
                "textDecoration": "none",
              },
              "to": "/home",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Test Teacher",
                "onClick": [Function],
                "style": Object {
                  "fontSize": 20,
                  "margin": 0,
                  "paddingLeft": 0,
                  "textDecoration": "none",
                  "textTransform": "none",
                },
              },
              "ref": null,
              "rendered": "Test Teacher",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Logout",
              "onClick": undefined,
              "style": Object {
                "marginLeft": "auto",
              },
            },
            "ref": null,
            "rendered": "Logout",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Navbar matches its snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar
    classes={
      Object {
        "appBar": Object {
          "root": Object {
            "display": "flex",
          },
        },
      }
    }
    person={
      Object {
        "teacher": Object {
          "name": "Test Teacher",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Toolbar)>
        <Link
          replace={false}
          style={
            Object {
              "textDecoration": "none",
            }
          }
          to="/home"
        >
          <WithStyles(Button)
            onClick={[Function]}
            style={
              Object {
                "fontSize": 20,
                "margin": 0,
                "paddingLeft": 0,
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
          >
            Test Teacher
          </WithStyles(Button)>
        </Link>
        <WithStyles(Button)
          style={
            Object {
              "marginLeft": "auto",
            }
          }
        >
          Logout
        </WithStyles(Button)>
      </WithStyles(Toolbar)>,
      "className": Object {
        "root": Object {
          "display": "flex",
        },
      },
      "color": "default",
      "id": "navbar",
      "position": "fixed",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            style={
              Object {
                "textDecoration": "none",
              }
            }
            to="/home"
          >
            <WithStyles(Button)
              onClick={[Function]}
              style={
                Object {
                  "fontSize": 20,
                  "margin": 0,
                  "paddingLeft": 0,
                  "textDecoration": "none",
                  "textTransform": "none",
                }
              }
            >
              Test Teacher
            </WithStyles(Button)>
          </Link>,
          <WithStyles(Button)
            style={
              Object {
                "marginLeft": "auto",
              }
            }
          >
            Logout
          </WithStyles(Button)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Button)
              onClick={[Function]}
              style={
                Object {
                  "fontSize": 20,
                  "margin": 0,
                  "paddingLeft": 0,
                  "textDecoration": "none",
                  "textTransform": "none",
                }
              }
            >
              Test Teacher
            </WithStyles(Button)>,
            "replace": false,
            "style": Object {
              "textDecoration": "none",
            },
            "to": "/home",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Test Teacher",
              "onClick": [Function],
              "style": Object {
                "fontSize": 20,
                "margin": 0,
                "paddingLeft": 0,
                "textDecoration": "none",
                "textTransform": "none",
              },
            },
            "ref": null,
            "rendered": "Test Teacher",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Logout",
            "onClick": undefined,
            "style": Object {
              "marginLeft": "auto",
            },
          },
          "ref": null,
          "rendered": "Logout",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Toolbar)>
          <Link
            replace={false}
            style={
              Object {
                "textDecoration": "none",
              }
            }
            to="/home"
          >
            <WithStyles(Button)
              onClick={[Function]}
              style={
                Object {
                  "fontSize": 20,
                  "margin": 0,
                  "paddingLeft": 0,
                  "textDecoration": "none",
                  "textTransform": "none",
                }
              }
            >
              Test Teacher
            </WithStyles(Button)>
          </Link>
          <WithStyles(Button)
            style={
              Object {
                "marginLeft": "auto",
              }
            }
          >
            Logout
          </WithStyles(Button)>
        </WithStyles(Toolbar)>,
        "className": Object {
          "root": Object {
            "display": "flex",
          },
        },
        "color": "default",
        "id": "navbar",
        "position": "fixed",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
              to="/home"
            >
              <WithStyles(Button)
                onClick={[Function]}
                style={
                  Object {
                    "fontSize": 20,
                    "margin": 0,
                    "paddingLeft": 0,
                    "textDecoration": "none",
                    "textTransform": "none",
                  }
                }
              >
                Test Teacher
              </WithStyles(Button)>
            </Link>,
            <WithStyles(Button)
              style={
                Object {
                  "marginLeft": "auto",
                }
              }
            >
              Logout
            </WithStyles(Button)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Button)
                onClick={[Function]}
                style={
                  Object {
                    "fontSize": 20,
                    "margin": 0,
                    "paddingLeft": 0,
                    "textDecoration": "none",
                    "textTransform": "none",
                  }
                }
              >
                Test Teacher
              </WithStyles(Button)>,
              "replace": false,
              "style": Object {
                "textDecoration": "none",
              },
              "to": "/home",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Test Teacher",
                "onClick": [Function],
                "style": Object {
                  "fontSize": 20,
                  "margin": 0,
                  "paddingLeft": 0,
                  "textDecoration": "none",
                  "textTransform": "none",
                },
              },
              "ref": null,
              "rendered": "Test Teacher",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Logout",
              "onClick": undefined,
              "style": Object {
                "marginLeft": "auto",
              },
            },
            "ref": null,
            "rendered": "Logout",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
